SU2_CFD/src/solvers/CNEMOEulerSolver.cpp

```diff
@@ -0,0 +1,5111 @@
+    SU2_MPI::Allreduce(MySurface_CFz_Inv, Surface_CFz_Inv, config->GetnMarker_Monitoring(), MPI_DOUBLE, MPI_SUM, MPI_COMM_WORLD);
+    SU2_MPI::Allreduce(MySurface_CMx_Inv, Surface_CMx_Inv, config->GetnMarker_Monitoring(), MPI_DOUBLE, MPI_SUM, MPI_COMM_WORLD);
+    SU2_MPI::Allreduce(MySurface_CMy_Inv, Surface_CMy_Inv, config->GetnMarker_Monitoring(), MPI_DOUBLE, MPI_SUM, MPI_COMM_WORLD);
+    SU2_MPI::Allreduce(MySurface_CMz_Inv, Surface_CMz_Inv, config->GetnMarker_Monitoring(), MPI_DOUBLE, MPI_SUM, MPI_COMM_WORLD);
+  }
+
+  delete [] MySurface_CL_Inv; delete [] MySurface_CD_Inv; delete [] MySurface_CSF_Inv;
+  delete [] MySurface_CEff_Inv;  delete [] MySurface_CFx_Inv;   delete [] MySurface_CFy_Inv;
+  delete [] MySurface_CFz_Inv;   delete [] MySurface_CMx_Inv;   delete [] MySurface_CMy_Inv;
+  delete [] MySurface_CMz_Inv;
+
+#endif
+
+  /*--- Update the total coefficients (note that all the nodes have the same value) ---*/
+  Total_CD           = AllBound_CD_Inv;
+  Total_CL           = AllBound_CL_Inv;
+  Total_CSF          = AllBound_CSF_Inv;
+  Total_CEff         = Total_CL / (Total_CD + EPS);
+  Total_CFx          = AllBound_CFx_Inv;
+  Total_CFy          = AllBound_CFy_Inv;
+  Total_CFz          = AllBound_CFz_Inv;
+  Total_CMx          = AllBound_CMx_Inv;
+  Total_CMy          = AllBound_CMy_Inv;
+  Total_CMz          = AllBound_CMz_Inv;
+  Total_CoPx         = AllBound_CoPx_Inv;
+  Total_CoPy         = AllBound_CoPy_Inv;
+  Total_CoPz         = AllBound_CoPz_Inv;
+  Total_CT           = AllBound_CT_Inv;
+  Total_CQ           = AllBound_CQ_Inv;
+  Total_CMerit       = Total_CT / (Total_CQ + EPS);
+  Total_CNearFieldOF = AllBound_CNearFieldOF_Inv;
+
+  /*--- Update the total coefficients per surface (note that all the nodes have the same value)---*/
+  for (iMarker_Monitoring = 0; iMarker_Monitoring < config->GetnMarker_Monitoring(); iMarker_Monitoring++) {
+    Surface_CL[iMarker_Monitoring]   = Surface_CL_Inv[iMarker_Monitoring];
+    Surface_CD[iMarker_Monitoring]   = Surface_CD_Inv[iMarker_Monitoring];
+    Surface_CSF[iMarker_Monitoring]  = Surface_CSF_Inv[iMarker_Monitoring];
+    Surface_CEff[iMarker_Monitoring] = Surface_CL_Inv[iMarker_Monitoring] / (Surface_CD_Inv[iMarker_Monitoring] + EPS);
+    Surface_CFx[iMarker_Monitoring]  = Surface_CFx_Inv[iMarker_Monitoring];
+    Surface_CFy[iMarker_Monitoring]  = Surface_CFy_Inv[iMarker_Monitoring];
+    Surface_CFz[iMarker_Monitoring]  = Surface_CFz_Inv[iMarker_Monitoring];
+    Surface_CMx[iMarker_Monitoring]  = Surface_CMx_Inv[iMarker_Monitoring];
+    Surface_CMy[iMarker_Monitoring]  = Surface_CMy_Inv[iMarker_Monitoring];
+    Surface_CMz[iMarker_Monitoring]  = Surface_CMz_Inv[iMarker_Monitoring];
+  }
+}
+
+void CNEMOEulerSolver::Momentum_Forces(CGeometry *geometry, CConfig *config) {
+
+  unsigned long iVertex, iPoint;
+  unsigned short iDim, iMarker, Boundary, Monitoring, iMarker_Monitoring;
+  su2double *Normal = nullptr, MomentDist[3] = {0.0,0.0,0.0}, *Coord, Area,
```

Comment:
We must really absolutely do something about this duplication, there is quite a bit of code here that is exactly the same as in CEulerSolver. This is an "inclusive we", I can help, and it will be a good opportunity to avoid some duplication in the incompressible solvers too.

Reply:
yes this is something Tim has already talked to me in slack, I am aware of the excess of repetition and it's already in my work plan to deal with it :)